tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/TelemetryValidator.cs

```diff
@@ -330,6 +331,24 @@ private Task Client_ProcessEventAsync(ProcessEventArgs arg)
             return Task.CompletedTask;
         }
 
+        private bool CheckRestartAnnouncement(dynamic json) {
+            if (json is JObject) {
+                try {
+                    var messageType = (string)json["MessageType"].Value;
```

Comment:
>(string) [](http://example.com/codeflow?start=38&length=8)

we could probably use here an "as string" instead of the cast to avoid an exception. same for the version